cli/dcoscli/node/main.py

```diff
@@ -65,12 +72,303 @@ def _cmds():
+            emitter.publish(snapshot_response)
+            return 0
+
+        emitter.publish(snapshot_response['status'])
+        return 0
+    else:
+        raise DCOSException('list_snapshots, status or cancel must be set')
+
+
+@snapshot_error
+def _do_request(url, method, timeout=3, stream=False, **kwargs):
+    """
+    make HTTP request
+
+    :param url: url
+    :type url: string
+    :param method: HTTP method, GET or POST
+    :type  method: string
+    :param timeout: HTTP request timeout, default 3 seconds
+    :type  timeout: integer
+    :param stream: stream parameter for requests lib
+    :type  stream: bool
+    :return: http response
+    :rtype: requests.Response
+    """
+    def _is_success(status_code):
+        # consider 400 and 503 to be successful status codes.
+        # API will return the error message.
+        if status_code in [200, 400, 503]:
+            return True
+        return False
+
+    # POST to snapshot api
+    client = mesos.DCOSClient()
+    url = client.get_dcos_url(url)
+    if method.lower() == 'get':
+        http_response = http.get(url, is_success=_is_success, timeout=timeout,
+                                 **kwargs)
+    elif method.lower() == 'post':
+        http_response = http.post(url, is_success=_is_success, timeout=timeout,
+                                  stream=stream, **kwargs)
+    else:
+        raise DCOSException('Unsupported HTTP method: ' + method)
+    return http_response
+
+
+def _do_snapshot_request(url, method, **kwargs):
+    """
+    Make HTTP request and expect a JSON response.
+
+    :param url: url
+    :type url: string
+    :param method: HTTP method, GET or POST
+    :type method: string
+    :return: snapshot JSON repsponse
+    :rtype: dict
+    """
+    http_response = _do_request(url, method, **kwargs)
+    return _read_http_response_body(http_response)
+
+
+def _read_http_response_body(http_response):
+    """
+    Get an requests HTTP response, read it and deserialize to json.
+
+    :param http_response: http response
+    :type http_response: requests.Response onject
+    :return: deserialized json
+    :rtype: dict
+    """
+    data = b''
+    try:
+        for chunk in http_response.iter_content(1024):
+            data += chunk
+        snapshot_response = util.load_jsons(data.decode('utf-8'))
+        return snapshot_response
+    except DCOSException:
+        raise
+
+
+def _snapshot_download(snapshot, location):
+    """
+    Download snapshot and put in the the current directory.
+
+    :param snapshot: snapshot file name.
+    :type snapshot: string
+    :return: exit code
+    :rtype: int
+    """
+
+    # make sure the requested snapshot exists
+    snapshot_size = 0
+    for available_snapshot in _get_snapshots_list():
+        if len(available_snapshot) != 2:
+            raise DCOSException('Incorrect snapshot response')
+
+        if available_snapshot[0] == snapshot:
+            snapshot_size = available_snapshot[1]
```

Comment:
added comments
we get a list of snapshots. A list of snapshots is a python list of tuples, where first element is a snapshot name and second is a snapshot size